src/course-unit/course-xblock/CourseXBlock.jsx

```diff
@@ -1,68 +1,104 @@
+import { useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import {
-  ActionRow, Card, Dropdown, Icon, IconButton,
+  ActionRow, Card, Dropdown, Icon, IconButton, useToggle,
 } from '@edx/paragon';
 import { EditOutline as EditIcon, MoreVert as MoveVertIcon } from '@edx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { scrollToElement } from '../../course-outline/utils';
 import messages from './messages';
 
-const CourseXBlock = ({ id, title }) => {
+const CourseXBlock = ({
+  id, title, unitXBlockActions, shouldScroll,
+}) => {
+  const currentRef = useRef(null);
```

Comment:
[optional]: I suggest a name indicating what exactly the ref refers to

```suggestion
  const courseXBlockElementRef = useRef(null);
```